### ISSUE TITLE
Error: Cannot obtain Metadata from http://address?wsdl

### ISSUE THREAD

**Issue by cperezortiz** [dotnet/wcf#1741]
Hi,

I've been trying to generate a soap client from the 'Connected Service' nuget package for visual studio that is based on this project.

The wsdl file is in a private host, I mean, it needs credentials, which are asked in the wizard, but at the end it fails scaffolding the code and throws out this error:

```
Scaffolding Code ...
Error:Error: Cannot obtain Metadata from http://address?wsdl 
If this is a Windows (R) Communication Foundation service to which you have access, please check that you have enabled metadata publishing at the specified address.  For help enabling metadata publishing, please refer to the MSDN documentation at http://go.microsoft.com/fwlink/?LinkId=65455.
WS-Metadata Exchange Error
    URI: http://address?wsdl
Metadata contains a reference that cannot be resolved: 'http://address?wsdl'.
One or more errors occurred. (An error occurred while sending the request. El parÃ¡metro no es correcto)
An error occurred in the tool.

Failed to generate service reference.
```

**Comment by zhenlan** [comment]
Sorry for running into this @cperezortiz. The CLI tool (svcutil) running under the cover of connected service does not support credentials yet. This is on our radar.

In the meanwhile, as a workaround, can you please save the wsdl as a file locally and load the file in the connected service to generate the code. Please make sure you have updated to the latest version of the WCF connected service.

**Comment by cperezortiz** [comment]
thanks for the feedback @zhenlan ,
I tried, downloading the wsdl locally, but looks like this one still needs to get connected to the address, maybe it is looking for some other files, i was told it may need its xds files, but i really don't know.

This is the error when loading it locally:

```
The document was understood, but it could not be processed.
  - The WSDL document contains links that could not be resolved.
  - There was an error downloading 'http://address?xsd=1'.
  - The request failed with HTTP status 401: Unauthorized.
If the service is defined in the current solution, try building the solution and adding the service reference again.
```

if i got those xsd files in the same directory as the wsdl you think that would help?

**Comment by hongdai** [comment]
@cperezortiz Could you send me the wsdl and xsd files? I'll attempt the repro and see if I can find a workaround for you.

**Comment by zhenlan** [comment]
@cperezortiz can you try to download the WSDL as a single WSDL file as a workaround? You can find more details from http://blogs.microsoft.co.il/idof/2011/09/17/whats-new-in-wcf-45-a-single-wsdl-file/.

**Comment by zhenlan** [comment]
We have made improvements of metadata download including authentication support in the latest update of the tool (released with VS 2017 v15.5 preview). Please give it a try and open new issues if you run into any other problems. Closing this issue for now.
